tests/test_object_model/test_site.py

```diff
@@ -1,99 +1,321 @@
-    assert isinstance(test_data.attrs.dem, DemModel)
-    assert isinstance(test_data.attrs.obs_station, Obs_stationModel)
-    assert isinstance(test_data.attrs.water_level, WaterLevelReferenceModel)
-    assert isinstance(test_data.attrs.water_level.other[0].height, UnitfulLength)
-    assert test_data.attrs.lat == 32.77
-    assert test_data.attrs.slr.vertical_offset.value == 0.6
-    assert test_data.attrs.fiat.exposure_crs == "EPSG:4326"
-    assert test_data.attrs.river[0].mean_discharge.value == 5000
 
+@pytest.fixture
+def test_sites(test_db, test_tomls):
+    test_sites = {toml_file.name: Site.load_file(toml_file) for toml_file in test_tomls}
+    return test_sites
 
-def test_read_site_toml_without_river(test_db):
-    test_toml = (
-        test_database / "charleston" / "static" / "site" / "site_without_river.toml"
-    )
 
-    assert test_toml.is_file()
+def test_loadFile_validFiles(test_tomls):
+    for toml_filepath in test_tomls:
+        Site.load_file(toml_filepath)
 
-    test_data = Site.load_file(test_toml)
 
-    assert isinstance(test_data.attrs.name, str)
-    assert isinstance(test_data.attrs.sfincs, SfincsModel)
-    assert isinstance(test_data.attrs.dem, DemModel)
-    assert isinstance(test_data.attrs.obs_station, Obs_stationModel)
-    assert test_data.attrs.lat == 32.77
-    assert test_data.attrs.slr.vertical_offset.value == 0.6
-    assert test_data.attrs.fiat.exposure_crs == "EPSG:4326"
+def test_loadFile_invalidFile_raiseFileNotFoundError(test_db):
+    with pytest.raises(FileNotFoundError):
+        Site.load_file("not_a_file.toml")
 
 
-@pytest.mark.skip(
-    reason="ID of observation station is None and site.toml cannot currently be saved"
-)
-def test_read_site_toml_with_multiple_rivers(test_db):
-    test_toml = test_database / "charleston" / "static" / "site" / "site.toml"
+def test_loadDict_validDict(test_dict):
+    test_site = Site.load_dict(test_dict)
+    assert isinstance(test_site.attrs, SiteModel)
+
+
+def get_required_attrs(cls):
+    required_attrs = []
+    for attr, type_annotation in cls.__annotations__.items():
+        if "Optional" not in str(type_annotation):
+            required_attrs.append(attr)
+    return required_attrs
+
+
+@pytest.mark.parametrize("attr, attr_type", list(SiteModel.__annotations__.items()))
+def test_loadDict_invalidDatatypes_raiseValidationError(attr, attr_type, test_dict):
+    if attr_type is int:
+        test_dict[attr] = "not an int"
+    else:
+        test_dict[attr] = 1
+    with pytest.raises(Exception):
+        Site.load_dict(test_dict)
+
+
+@pytest.mark.parametrize("removed_key", get_required_attrs(SiteModel))
+def test_loadDict_missingRequiredAttrs_raiseValidationError(test_dict, removed_key):
+    test_dict.pop(removed_key)
+    # Validation errors dont have an __eq__ method, so we cant use pytest.raises(ValidationError)
+    with pytest.raises(Exception) as e_info:
+        Site.load_dict(test_dict)
+    assert removed_key in str(e_info)
```

Comment:
If you still want to check if its a validation error, e_info.typename should be 'validationerror' so you could assert on that